library/bignum_mod.h

```diff
@@ -2,6 +2,61 @@
  *  Modular bignum functions
  *
  * This module implements operations on integers modulo some fixed modulus.
+ *
+ * The functions in this module obey the following conventions unless
+ * explicitly indicated otherwise:
+ *
+ * - **Modulus parameters**: the modulus is passed as a pointer to a structure
+ *   of type #mbedtls_mpi_mod_modulus. The structure must be setup with an
+ *   array of limbs storing the bignum value of the modulus. Unless otherwise
+ *   specified, the modulus is called \p N and is input-only.
+ * - **Bignum parameters**: Bignums are passed as pointers to an array of
+ *   limbs or to a #mbedtls_mpi_mod_residue structure. A limb has the type
+ *   #mbedtls_mpi_uint. Residues must be initialized before use, and must be
+ *   associated with the modulus \p N. Unless otherwise specified:
+ *     - Bignum parameters called \p A, \p B, ... are inputs and are not
+ *       modified by the function. These will have the type
+ *       #mbedtls_mpi_mod_residue.
```

Comment:
This is not addressed. Please document that functions that take a parameter of type `const mbedtls_mpi_mod_residue *` (or for that matter `const mbedtls_mpi_mod_modulus *`) don't modify the value (even though C would allow modifying the limb array).

Reply:
Apologies, I've now documented this for both modulus and residue.